engine/crates/engine-v2/src/sources/graphql/federation.rs

```diff
@@ -179,23 +189,22 @@ impl<'ctx> FederationEntityRequest<'ctx> {
             })
             .map_err(|err| format!("Failed to serialize query: {err}"))?;
 
-            execute_subgraph_request(ctx, span, headers, Bytes::from(body), ingester).await
+            execute_subgraph_request(ctx, headers, Bytes::from(body), ingester).await
         }
-        .instrument(span_clone)
+        .instrument(span)
         .await
     }
 }
 
 async fn cache_fetches<'ctx, R: Runtime>(
-    ctx: ExecutionContext<'ctx, R>,
-    endpoint: GraphqlEndpoint<'ctx>,
+    ctx: &mut SubgraphContext<'ctx, R>,
     headers: &http::HeaderMap,
     representations: Vec<Box<RawValue>>,
     additional_scopes: &[String],
 ) -> CacheFetchOutcome {
     let fetches = representations
         .iter()
-        .map(|repr| cache_fetch(ctx, endpoint, headers, repr, additional_scopes));
+        .map(|repr| cache_fetch(ctx, ctx.endpoint, headers, repr, additional_scopes));
```

Comment:
oh yes should, good catch

Reply:
ah no, nevermind, the context must be shared here as we run those futures concurrently. So I'm actually passing `ExecutionContext` rather than the `SubgraphContext`